modules/heat_transfer/src/bcs/GapPerfectConductance.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GapPerfectConductance.h"
+
+registerMooseObject("HeatTransferApp", GapPerfectConductance);
+
+InputParameters
+GapPerfectConductance::validParams()
+{
+  InputParameters params = IntegratedBC::validParams();
+  params.addClassDescription("Enforces equal temperatures across the gap.");
+  params.addRequiredCoupledVar("gap_distance", "Distance across the gap.");
+  params.addRequiredCoupledVar("gap_temp", "Temperature on the other side of the gap.");
+  params.addParam<Real>("penalty", 1, "Penalty value to be applied to the constraint.");
```

Comment:
is 1 a reasonable penalty? It seems a little weak to me. Maybe 1e3?

Reply:
That's problem dependent, but it is larger than 1 for coupled problems, I'll increase the default.